### PR TITLE
fix(svg-editor): delete sub-selected vertices in path edit mode (#880)

### LINKED ISSUE #880: @grida/svg-editor: Delete/Backspace in path edit mode removes the whole element instead of the selected vertices
## Summary

In `@grida/svg-editor`, while in **path edit mode** (`state.mode === "edit-content"` / vector-edit), selecting a vertex (or a segment/tangent sub-selection) and pressing <kbd>Delete</kbd> / <kbd>Backspace</kbd> **removes the entire path element** from the document, instead of deleting only the sub-selected geometry from the path's `d` (or `points`).

## Steps to reproduce

1. Open the svg-editor (the playground at `editor/app/(tools)/(packages)/packages/[%40grida]/svg-editor`, or a workspace `.svg` tab in desktop).
2. Select a `<path>` (or `<polyline>` / `<polygon>`) and enter path edit mode (double-click, or `Enter`).
3. Click a single vertex to sub-select it.
4. Press <kbd>Delete</kbd> or <kbd>Backspace</kbd>.

**Expected:** the sub-selected vertex is removed; the path geometry is rewritten to reflect the smaller shape; the element survives and stays in edit mode.

**Actual:** the whole element is detached from the document as a single undoable `remove` step.

## Root cause

`Delete` / `Backspace` are bound unconditionally to the `selection.remove` command, and that handler is the **only structural-mutation handler in the file without a mode guard**.

Bindings — no mode distinction (`packages/grida-svg-editor/src/keymap/defaults.ts:43`):

```ts
{ keybinding: kb(KeyCode.Backspace), command: "selection.remove" },
{ keybinding: kb(KeyCode.Delete), command: "selection.remove" },
```

Handler (`packages/grida-svg-editor/src/commands/defaults.ts:64`):

```ts
reg.register("selection.remove", () => {
  if (editor.state.selection.length === 0) return false;
  editor.commands.remove(); // detaches the whole node subtree
  return true;
});
```

Every sibling structural command (`selection.group`, `selection.duplicate`, `selection.ungroup`, `clipboard.*`, …) self-guards with `if (editor.state.mode !== "select") return false;`. `selection.remove` does not.

Entering vector-edit flips `state.mode` to `"edit-content"` but **does not clear `state.selection`** — the path node stays selected (it is the node under edit, used for HUD chrome and exit). See `_do_enter_vector_edit` (`packages/grida-svg-editor/src/dom.ts:4255`).

So in edit-content mode `selection.length` is still `1` (the path); the guard-less handler runs `editor.commands.remove()` → `remove_selection("remove")` (`packages/grida-svg-editor/src/core/editor.ts:1856`), which prunes and detaches the path subtree.

## This was already the planned design — it just never shipped

The package's own keybindings tracker flags exactly this. `packages/grida-svg-editor/docs/keybindings.md:73`:

> | Delete | `Delete` or `Backspace` | … | `[x]` | `selection.remove` | shipped; **future routing inside the handler** |

And the policy-class layer already declares `delete-vertex` as a first-class intent, with chosen policies (`restrict` for vertex-chain — polygon ≥ 3, line == 2; `bake` for path):

- `packages/grida-svg-editor/src/core/policy-class/types.ts:65` and `:83` (intent vocabulary)
- `packages/grida-svg-editor/src/core/policy-class/index.ts:117`, `:160`, `:202` (solution space + chosen policy)
- spec: `docs/wg/feat-svg-editor/glossary/policy-class.md:471`

…but nothing invokes it from the keyboard path. There is **no** `vector.delete-vertex` command registered and **no** keybinding pointing at one.

## Prior art (reference for the fix)

The main canvas editor (`editor/grida-canvas`) already implements the correct behavior in path-edit mode: delete selected tangents → segments → vertices, then clear the sub-selection — `editor/grida-canvas/reducers/document.reducer.ts:2408`. The svg-editor should mirror this shape.

## Proposed fix (two parts)

**1. Stop the over-delete.** Add the standard mode guard to `selection.remove` so it only deletes elements in `"select"` mode:

```ts
reg.register("selection.remove", () => {
  if (editor.state.mode !== "select") return false; // chain falls through in edit-content
  if (editor.state.selection.length === 0) return false;
  editor.commands.remove();
  return true;
});
```

On its own this makes Delete a no-op in path-edit mode — correct over deleting the element, but incomplete.

**2. Wire vertex deletion.** Register a `vector.delete-vertex` command (gated on `mode === "edit-content"` + a non-empty vector sub-selection) and add chained `Delete` / `Backspace` rows in `keymap/defaults.ts`. The keymap supports same-key chains via `false`-returns (see the `content.enter` → `hierarchy.enter` chain). The command should remove the selected tangents/segments/vertices from the active `VectorEditSession` and project back via `apply_session_d`, honoring the policy-class `delete-vertex` verdict (e.g. `restrict` refuses dropping a polygon below 3 vertices).

## Affected files

- `packages/grida-svg-editor/src/commands/defaults.ts` — guard + new handler
- `packages/grida-svg-editor/src/keymap/defaults.ts` — chained Delete/Backspace rows
- `packages/grida-svg-editor/src/core/vector-edit/*` — delete-on-session implementation
- `packages/grida-svg-editor/docs/keybindings.md` — update the Delete row status

## Acceptance criteria

- [ ] In `"edit-content"` mode with a vertex/segment sub-selection, Delete/Backspace removes only the sub-selected geometry; the element survives and stays in edit mode.
- [ ] Deletion is a single undo step; undo restores the prior geometry and sub-selection.
- [ ] Policy is respected (e.g. a triangle `<polygon>` refuses to drop below 3 vertices per the `restrict` policy).
- [ ] In `"select"` mode, Delete/Backspace still removes the selected element(s) (unchanged).
- [ ] With an empty sub-selection in edit mode, Delete is a no-op (no element deletion).
- [ ] Tests added under `packages/grida-svg-editor/__tests__/` (e.g. `vector-edit-delete.test.ts`).

### PR DESCRIPTION
**Pull request by softmarshmallow**
Closes #880.

## Problem

In `@grida/svg-editor`, while in **path edit mode** (`state.mode === "edit-content"`), selecting a vertex and pressing <kbd>Delete</kbd> / <kbd>Backspace</kbd> **removed the entire path element** instead of the sub-selected geometry.

Root cause: `Delete` / `Backspace` were bound unconditionally to `selection.remove`, the **only structural-mutation handler without a mode guard**. Entering vector-edit flips the mode but keeps the path node selected, so the guard-less handler detached the whole subtree.

## Fix (two parts)

**1. Stop the over-delete.** Guard `selection.remove` on `select` mode — the same shape every sibling structural command (`selection.group` / `duplicate` / `ungroup` / `clipboard.*`) already uses.

**2. Wire vertex deletion.** A new `vector.delete-vertex` path, mode-routed ahead of `selection.remove` via the same `false`-return keymap chain the `Enter` overload (`content.enter` → `hierarchy.enter`) uses:

- **`PathModel.deleteSubSelection(sel)`** — pure geometry, mirroring the main canvas editor's `__self_delete_vector_network_selection` (tangents → segments → vertices via `@grida/vn`'s delete ops; non-reconnecting, matching the reference).
- **`delete_vector_subselection()`** — pure core that honors the policy-class `delete-vertex` verdict: `restrict` for vertex-chain (polygon ≥ 3, polyline ≥ 2, line keeps 2) and `bake` for path.
- **`commands.delete_vector_selection()` + `set_vector_delete_driver`** — surface driver, symmetric to `set_vector_selection` (the session is surface-owned).
- **Surface handler** routes the write through the existing `vector_geometry_step` chokepoint, so native-attr writeback and promote-to-`<path>` are handled uniformly. One undo step; undo restores **both** geometry and the deleted sub-selection; the element survives and stays in edit mode.

## Acceptance criteria

- [x] Edit-content + sub-selection: Delete/Backspace removes only the sub-selected geometry; element survives, stays in edit mode.
- [x] Single undo step restores prior geometry **and** sub-selection.
- [x] Policy respected — a triangle `<polygon>` refuses to drop below 3 vertices.
- [x] Select mode: Delete/Backspace still removes the element (unchanged).
- [x] Empty sub-selection in edit mode: no-op (no element deletion).
- [x] Tests added under `packages/grida-svg-editor/__tests__/`.

## Testing

- New `vector-edit-delete.test.ts` — 22 tests across all three layers (geometry, policy decision, editor + keymap routing).
- Full package suite: **1498 passing**; `tsc --noEmit`, oxlint, oxfmt all clean.
- Manual E2E case added: `test/svg-editor-vector-delete.md` (the DOM surface can't mount in jsdom; the headless tests cover the decision + geometry + routing).

## Reviewer notes

- Deletion mirrors the main editor's **non-reconnecting** `vn.deleteVertex` (a deleted interior vertex's neighbours are not bridged) — the issue directed "mirror this shape". The interior-vertex test pins this explicitly; reconnection would be a separate cross-editor enhancement.
- `min_vertices` (the per-tag numeric minimums) lives in `delete.ts` for now — single call site. Flagged as latent debt to pull into the policy-class layer when a second consumer (close/open-shape) appears.

## Summary by CodeRabbit

* **New Features**
  * Added mode-aware Delete/Backspace behavior for vector path editing via a new `vector.delete-vertex` action.
  * In path-edit mode, Delete/Backspace deletes only the selected sub-elements (vertices/segments/tangents) while keeping the path element in place and clearing the sub-selection.
* **Bug Fixes**
  * Enforces restrict/bake delete policies consistently, prevents accidental element removal in non-select modes, and treats deletion as a single undo step.
* **Documentation**
  * Updated keybinding docs to describe the new Delete routing and path-edit deletion rules.
* **Tests**
  * Added coverage for sub-selection deletion across key edge cases and mode routing behavior.

### CODE REVIEW

File: packages/grida-svg-editor/src/core/vector-edit/model.ts [hunk 1]
```diff
@@ -530,6 +530,72 @@ export class PathModel {
...
+    for (const vertex_idx of [...sel.vertices].sort((a, b) => b - a)) {
+      if (vertex_idx >= 0 && vertex_idx < vne.value.vertices.length) {
+        vne.deleteVertex(vertex_idx);
+      }
+    }
+
+    // `next_network` is this method's own clone — hand it to PathModel
+    // directly (no second clone) with fresh verb-less meta, matching the
+    // sibling mutators (`translateVertex`, `splitSegment`, …).
+    const meta: SegmentMeta[] = vne.value.segments.map(() => ({}));
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-svg-editor/src/core/vector-edit/model.ts]: Fixed in 4fac41679 — `deleteSubSelection` now preserves the original meta verbatim for tangent-only deletes (segment array shape is unchanged), so untouched arcs/H/V/Z keep their verbs; the edited tangent's owning segment/arc auto-invalidates at emit time (zeroed tangents → L, mutated arc → C). Only topology-changing deletes (segment/vertex removal) reset to verb-less meta. Added a test pinning that an untouched H survives while the cubic whose tangents were deleted demotes to L.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: test/svg-editor-vector-delete.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+     rendered shape update to the smaller path; the element is still the same
+     `<path id="p">` (NOT detached from the document), still in edit mode, and
+     the sub-selection is now empty.
+4. Press <kbd>Cmd/Ctrl+Z</kbd> once.
+   - Expected: a single undo restores the prior geometry **and** re-selects the
+     vertex that was deleted.
+5. Repeat step 3 with <kbd>Backspace</kbd> — identical behavior to
+   <kbd>Delete</kbd>.
+6. With nothing sub-selected (click empty space inside edit mode to clear the
+   sub-selection), press <kbd>Delete</kbd>.
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply test/svg-editor-vector-delete.md]: Fixed in 07f8a3a38 — Step 6 now reads "click empty space to clear the sub-selection" (dropped the redundant "inside edit mode").

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_

File: packages/grida-svg-editor/src/core/vector-edit/model.ts [hunk 3]
```diff
@@ -530,6 +530,93 @@ export class PathModel {
...
+      if (seg_idx >= 0 && seg_idx < vne.value.segments.length) {
+        vne.deleteSegment(seg_idx);
+        topology_changed = true;
+      }
+    }
+
+    // 3. vertices — descending (deleteVertex reindexes higher vertices on splice).
+    for (const vertex_idx of [...sel.vertices].sort((a, b) => b - a)) {
+      if (vertex_idx >= 0 && vertex_idx < vne.value.vertices.length) {
+        vne.deleteVertex(vertex_idx);
```
[the remark this answers is not in the document]
Comment (softmarshmallow) [reply packages/grida-svg-editor/src/core/vector-edit/model.ts]: Fixed in 9a08de9ee — both the segment and vertex delete loops now dedupe via `new Set(...)` before the descending sort, so a duplicate index can't over-delete after reindexing. This also aligns the loops with the policy gate's `new Set(vertices).size` unique-count reasoning. Added a test pinning that `[1, 1]` deletes the same geometry as `[1]`.

_🤖 Addressed by [Claude Code](https://claude.com/claude-code)_
